packages/suite/src/utils/suite/protocol.ts

```diff
@@ -0,0 +1,35 @@
+import { PROTOCOL_SCHEME } from '@suite-constants/protocol';
+import { parseQuery, parseUri } from './parseUri';
+
+export type CoinProtocolInfo = {
+    scheme: PROTOCOL_SCHEME;
+    address: string;
+    amount?: number;
+};
+
+export const isProtocolScheme = (scheme: string): scheme is PROTOCOL_SCHEME =>
+    Object.values(PROTOCOL_SCHEME).includes(scheme as PROTOCOL_SCHEME);
+
+export const getProtocolInfo = (uri: string): CoinProtocolInfo | null => {
+    const url = parseUri(uri);
+    if (!url) return null;
```

Comment:
nit: I think we agreed on not using one line ifs on previous code improvements 👀

Reply:
1e7b694621bbadbc30dbac7245b55b430a214043